http/client/client.go

```diff
@@ -16,6 +16,33 @@ type Requester interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
+// TimedClient instruments a request. It implements Requester.
+type TimedClient struct {
+	client    Requester
+	collector instrument.Collector
+}
+
+// CtxTimedOperationNameKey specifies the operation name location within the context
+// for instrumentation.
+const CtxTimedOperationNameKey = "op"
```

Comment:
Will this cause problems if two timing contexts are nested?

Reply:
Shouldn't, because the context is tied to the request. (If you nest `TimedClient`s it will just measure it twice for the same op)